components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/model/TokenRequest.java

```diff
@@ -103,15 +103,15 @@ public Builder scopes(List<String> scopes) {
             return this;
         }
 
-        public Builder addAdditionalHeader(String key, String[] value) {
+        public Builder addAdditionalHeader(Header header) {
 
-            this.additionalHeaders.put(key, value);
+            this.additionalHeaders.add(header);
             return this;
         }
 
-        public Builder addAdditionalParam(String key, String[] value) {
+        public Builder addAdditionalParam(Param param) {
```

Comment:
Let's keep this also as name, value pair

Reply:
This is updated with bf1b5d4d66da5d1808354cc3dc663adcbed6359d